lib/engines/content_object_store/app/controllers/content_object_store/content_block/editions_controller.rb

```diff
@@ -1,95 +1,35 @@
-      @schemas = ContentObjectStore::ContentBlock::Schema.all
-    else
-      @schema = ContentObjectStore::ContentBlock::Schema.find_by_block_type(params[:block_type].underscore)
-      @form = ContentObjectStore::ContentBlock::EditionForm::Create.new(
-        content_block_edition: ContentObjectStore::ContentBlock::Edition.new,
-        schema: @schema,
-      )
-    end
+    @content_block_document = ContentObjectStore::ContentBlock::Document.find(params[:id])
+    @schema = ContentObjectStore::ContentBlock::Schema.find_by_block_type(@content_block_document.block_type)
+    @form = ContentObjectStore::ContentBlock::EditionForm.new(
+      content_block_edition: @content_block_document.latest_edition,
+      schema: @schema,
+    )
   end
 
   def create
     @schema = ContentObjectStore::ContentBlock::Schema.find_by_block_type(block_type_param)
-
-    new_edition = ContentObjectStore::CreateEditionService.new(@schema).call(edition_params)
-
-    redirect_to content_object_store.review_content_object_store_content_block_edition_path(new_edition)
-  rescue ActiveRecord::RecordInvalid => e
-    @form = ContentObjectStore::ContentBlock::EditionForm::Create.new(content_block_edition: e.record, schema: @schema)
-    render :new
-  end
-
-  def review
-    @content_block_edition = ContentObjectStore::ContentBlock::Edition.find(params[:id])
-  end
-
-  EDIT_FORM_STEPS = {
-    edit_block: "edit_block",
-    review_links: "review_links",
-    schedule_publishing: "schedule_publishing",
-  }.freeze
-
-  def edit
-    step = params[:step]
-    @content_block_edition = ContentObjectStore::ContentBlock::Edition.find(params[:id])
-    @schema = ContentObjectStore::ContentBlock::Schema.find_by_block_type(@content_block_edition.document.block_type)
-
-    case step
-    when EDIT_FORM_STEPS[:edit_block]
-      edit_block
-    when EDIT_FORM_STEPS[:review_links]
-      review_links
-    when EDIT_FORM_STEPS[:schedule_publishing]
-      schedule_publishing
-    end
-  end
-
-  def edit_block
-    @form = ContentObjectStore::ContentBlock::EditionForm::Update.new(
-      content_block_edition: @content_block_edition, schema: @schema, edition_to_update_id: @content_block_edition.id,
-    )
-  end
-
-  def review_links
-    @content_block_document = @content_block_edition.document
-    @edition_params = edition_params
-
     new_edition = ContentObjectStore::ContentBlock::Edition.new(edition_params)
-    new_edition.document.id = @content_block_document.id
+    new_edition.document_id = params[:id]
+    new_edition.document.assign_attributes(edition_params[:document_attributes].except(:block_type))
 
-    if new_edition.valid?
-      @host_content_items = ContentObjectStore::GetHostContentItems.by_embedded_document(
-        content_block_document: @content_block_document,
-      )
-
-      render :review_links
+    if new_edition.valid? && new_edition.document.valid?
+      new_edition.save!
```

Comment:
so here we create a new Edition before the user has actually got to the end of the form? That means we go back to having unseen Draft Content Block Editions around if they don't complete the form?

Reply:
ah sorry I think I'm getting confused by the diff

Reply:
is this `create` method is used when both creating the first Edition and when adding subsequent Editions?

Reply:
I think while this is more strictly RESTful it might miss the realities of the publishing workflow - I think practically there's a big difference between creating a new Edition on a new Document and adding new Editions to an existing Document, so it would be helpful to keep the `edit` step separate to the `create` step. I also would prefer us not to go back to creating orphaned Draft Editions if someone abandons a form.

Reply:
Yeah, I think we'll have to if we want to preview editions. It's not ideal, but I can't really see a way round it. The happy path _should_ be that users continue through the flow, so the issue of there being draft editions hanging around should be minimal

Reply:
ah that's a shame that we have to create the Edition ... I guess if it does become a problem we can have a task to tidy them